src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1135,29 +1134,29 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				windowsShellArgs = true;
 				// If we don't have a cwd, then the terminal uses the home dir.
 				const userHome = await this._pathService.userHome();
-				if (basename === 'cmd.exe' && ((options.cwd && isUNC(options.cwd)) || (!options.cwd && isUNC(userHome.fsPath)))) {
-					return undefined;
-				}
-				if ((basename === 'powershell.exe') || (basename === 'pwsh.exe')) {
-					if (!shellSpecified) {
+				if (basename === 'cmd.exe') {
+					if ((options.cwd && isUNC(options.cwd)) || (!options.cwd && isUNC(userHome.fsPath))) {
+						return undefined;
+					}
+					if (!shellArgsSpecified) {
+						toAdd.push('/d', '/c');
+					}
+				} else if ((basename === 'powershell.exe') || (basename === 'pwsh.exe')) {
+					if (!shellArgsSpecified) {
 						toAdd.push('-Command');
 					}
 				} else if ((basename === 'bash.exe') || (basename === 'zsh.exe')) {
 					windowsShellArgs = false;
-					if (!shellSpecified) {
+					if (!shellArgsSpecified) {
 						toAdd.push('-c');
 					}
 				} else if (basename === 'wsl.exe') {
-					if (!shellSpecified) {
+					if (!shellArgsSpecified) {
 						toAdd.push('-e');
 					}
-				} else {
-					if (!shellSpecified) {
-						toAdd.push('/d', '/c');
-					}
```

Comment:
I'm not sure if we should keep `/d /c` as the default for Windows like before (like what we did `-c` for other OSs)

My opinion is default to `-c` for all OSs, since most shells are using it